### ISSUE TITLE
redisson don't detect if new sentinel added to sentinel group.

### ISSUE THREAD

**Issue by bhaskarroy19** [redisson/redisson#2067]
### Redis version
5.*
### Redisson version
3.10.5

**Comment by mrniko** [comment]
Could you please set logging level to `TRACE` for `org.redisson` and share log output?

**Comment by bhaskarroy19** [comment]
I am debugging it in intelliJ and not using any logger config. How can I set it?

**Comment by bhaskarroy19** [comment]
I set the logging level to TRACE but it not showing anything. 
I monitored it at application debugging and have seen that the  sentinelServers not changing as per sentinel down or up.

**Comment by bhaskarroy19** [comment]
So, I debugged it again and have seen that 
1.  When a sentinel node down no changes happened there. sentinel connection manager not showing any sentinel node down info.
2. But If I add a new sentinel then the new node information is showing in the sentinel connection manager. 
So, If earlier I had 3 sentinels as S1, S2, S3, and S3 went down, the connection manager still showing S1, S2, S3 and number of sentinel server count stays as 3.
Now if I add a new sentinel S4 then it added to connection manager and the number of sentinel server count increased to 4 and servers showing as S1, S2, S3, S4. 
and it is growing on continuously as I added a new sentinel even actually my active sentinel server count still stays 3, like, 
3, 4, 5, .......
S1, S2, S3, S4, S5, .........
So, basically, it holds stale information about active servers.
can you please tell why this is happening? and what will be an impact on production?

**Comment by mrniko** [comment]
> I am debugging it in intelliJ and not using any logger config. How can I set it?

You need to drop slf4j compatible logger in classpath like logback.

**Comment by mrniko** [comment]
> I monitored it at application debugging and have seen that the sentinelServers not changing as per sentinel down or up.

This is why I'm asking to provide log output

**Comment by bhaskarroy19** [comment]
My sentinel setting: 
127.0.0.1:5000, 127.0.0.1:5001, 127.0.0.1:5002
master-slave:
127.0.0.1:6379  (slave)
127.0.0.1:6380 (slave)
127.0.0.1:6381 (master)
Hi, I am attaching a log file which logs everything as TRACE level when I did the following activity:
1. stopped the sentinel with port 5002.
2. notify other sentinels about 5002 no more present.
3. wait
4. up another sentinel with port 5003
5. sentinel got automatically notified.
6. wait
7. again down the new Sentinel 5003.
8. notified other sentinels about the event.
9. wait.
10. up the old sentinel with 5002 port.
11. did a master-slave switch-over.
[audit.log](https://github.com/redisson/redisson/files/3121037/audit.log)

**Comment by mrniko** [comment]
I confirm this bug -  broken Sentinels are not discovered.

**Comment by bhaskarroy19** [comment]
I re-tested it. Can you check another log file?. 
[redisson.log](https://github.com/redisson/redisson/files/3121164/redisson.log)
The testing sequence is the same as previous. 
This time in the log I am able to see the changes. Means, when I down 5002, then "SENTINEL sentinels mymaster" replying with *1 and it not showing 5002 sentinel. 
And when I add sentinel 5003, then "SENTINEL sentinels " command replayong *2 and showing two other sentinels where one was newly added 5003 sentinel.
Means, It working fine this time. 
But still one thing is wrong there in sentinels values in connectionManger.  So, At the time of debugging I was cheching the values of sentinels in connectionManager of Radisson, and it was showing sentinel servers count wrongly. It showing stale sentinel information along with new sentinel.  
I already described it which is as below: 

So, If earlier I had 3 sentinels as S1, S2, S3, and S3 went down, the connection manager still showing S1, S2, S3 and number of sentinel server count stays as 3.
Now if I add a new sentinel S4 then it added to connection manager and the number of sentinel server count increased to 4 and servers showing as S1, S2, S3, S4.
and it is growing on continuously as I added a new sentinel even actually my active sentinel server count still stays 3, like,
3, 4, 5, .......
S1, S2, S3, S4, S5, .........
So, basically, it holds stale information about active servers.

**Comment by mrniko** [comment]
I fixed the problem when offline sentinels are still present in list of sentinels.

Please find updated module below
[redisson-3.10.7-SNAPSHOT.jar.zip](https://github.com/redisson/redisson/files/3121285/redisson-3.10.7-SNAPSHOT.jar.zip)

**Comment by bhaskarroy19** [comment]
Hi, the new jar is not even showing connected sentinels in sentinels under connectionManager. the map sentinels empty under connectionManager. And also it stuck at the time of master switchover. 
redisson-3.10.5 has only one problem that it showing stale sentinel values in map sentinels under connectionManager. You just need to resolve that, nothing else.

**Comment by mrniko** [comment]
Fixed

**Comment by bhaskarroy19** [comment]
which version will contain the resolved code?

**Comment by mrniko** [comment]
@bhaskarroy19 

3.10.7

**Comment by bhaskarroy19** [comment]
Great. When will you release it to maven repository?

**Comment by mrniko** [comment]
@bhaskarroy19 

It's already there

**Comment by bhaskarroy19** [comment]
I am only getting till 3.10.6. 
Not able to find Redisson-3.10.7 there.

**Comment by mrniko** [comment]
@bhaskarroy19 

http://central.maven.org/maven2/org/redisson/redisson/3.10.7/

**Comment by bhaskarroy19** [comment]
Thanks. Even in intelliJ when I am entering 3.10.7, is able to resolve the dependency.  
Can you please look here [https://mvnrepository.com/artifact/org.redisson/redisson]

**Comment by bhaskarroy19** [comment]
Great. It is working now.
Can you tell me how you get the current active sentinels list? 
earlier you said, you call "Sentinel sentinels" command in every interval. 
I have seen that even though I didn't call "sentinel reset mymaster" to other sentinels, but you are able to get current active sentinels. 
But at the same time my "sentinel sentinels" command to active sentinels showing old sentinel value.
